src/main/java/sirius/biz/storage/layer1/ObjectStorageSpace.java

```diff
@@ -246,6 +249,28 @@ public void upload(String objectId, File file) {
     protected abstract void storePhysicalObject(String objectKey, File file, ByteBlockTransformer transformer)
             throws IOException;
 
+    /**
+     * Copies the given object to the given target object key in the current storage space.
+     *
+     * @param sourceObjectKey    the source object key to copy from
+     * @param targetObjectKey    the target object key to copy to
+     * @param targetStorageSpace the target storage space to copy to
+     * @throws IOException in case of an IO error
+     */
+    public abstract void copyPhysicalObject(String sourceObjectKey, String targetObjectKey, String targetStorageSpace)
```

Comment:
do we really need a targetStorageSpace - I don't think we can optimize copying accross multiple spaces - but each impl would have to take care of it.

or we handle the "not same case" in the base class. we could provide a "hand rolled copy" and only optimize the s3 case and otherwise always call super.copy... 

Also, maybe we call this "duplicate" to better show what it does(?)